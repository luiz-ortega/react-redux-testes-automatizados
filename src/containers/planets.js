import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchPlanets } from '../store/actions';
import { userPlanetList } from '../store/selectors/planets-selectors';
import Filters from './filters';
import Deck from '../components/planets/deck/main';

export default function Planets() {
  const dispatch = useDispatch();
  const { loading, error } = useSelector(store => store.planets);
  const list = useSelector(userPlanetList);

  useEffect(() => {
    dispatch(fetchPlanets());
  }, [dispatch]);

  return (
    <>
      {error ? <h4>So sorry...</h4> : null}
      {loading ? <h4>Loading...</h4> : null}
      {!loading ? (
        <>
          <div className="mb-5">
            <Filters />
          </div>
          <Deck dataSource={list} />
        </>
      ) : null}
    </>
  );
}
