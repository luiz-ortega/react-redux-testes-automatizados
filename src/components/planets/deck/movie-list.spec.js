import React from 'react';
import { render } from '@testing-library/react';
import MovieList from './movie-list';
import planets from '../../../../test/stubs/planets';

function renderComponent() {
  return render(<MovieList films={planets.results[0].films} />);
}

describe('Component: MovieList', () => {
  describe('Displays all elements', () => {
    it('should match snapshot', () => {
      const { baseElement } = renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
  });
});
