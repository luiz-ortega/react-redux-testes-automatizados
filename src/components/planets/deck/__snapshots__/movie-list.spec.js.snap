// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MovieList Displays all elements should match snapshot 1`] = `
<body>
  <div>
    <ul
      class="border border-gray-200 rounded-md divide-y divide-gray-200"
    >
      <li
        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
      >
        <div
          class="w-0 flex-1 flex items-center"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="ml-2 flex-1 w-0 truncate"
          >
            /api/films/1/
          </span>
        </div>
        <div
          class="ml-4 flex-shrink-0"
        >
          <a
            class="font-medium text-indigo-600 hover:text-indigo-500"
            href="/api/films/1/"
          >
            Visit
          </a>
        </div>
      </li>
      <li
        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
      >
        <div
          class="w-0 flex-1 flex items-center"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="ml-2 flex-1 w-0 truncate"
          >
            /api/films/3/
          </span>
        </div>
        <div
          class="ml-4 flex-shrink-0"
        >
          <a
            class="font-medium text-indigo-600 hover:text-indigo-500"
            href="/api/films/3/"
          >
            Visit
          </a>
        </div>
      </li>
      <li
        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
      >
        <div
          class="w-0 flex-1 flex items-center"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="ml-2 flex-1 w-0 truncate"
          >
            /api/films/4/
          </span>
        </div>
        <div
          class="ml-4 flex-shrink-0"
        >
          <a
            class="font-medium text-indigo-600 hover:text-indigo-500"
            href="/api/films/4/"
          >
            Visit
          </a>
        </div>
      </li>
      <li
        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
      >
        <div
          class="w-0 flex-1 flex items-center"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="ml-2 flex-1 w-0 truncate"
          >
            /api/films/5/
          </span>
        </div>
        <div
          class="ml-4 flex-shrink-0"
        >
          <a
            class="font-medium text-indigo-600 hover:text-indigo-500"
            href="/api/films/5/"
          >
            Visit
          </a>
        </div>
      </li>
      <li
        class="pl-3 pr-4 py-3 flex items-center justify-between text-sm"
      >
        <div
          class="w-0 flex-1 flex items-center"
        >
          <svg
            class="h-5 w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4v16M17 4v16M3 8h4m10 0h4M3 12h18M3 16h4m10 0h4M4 20h16a1 1 0 001-1V5a1 1 0 00-1-1H4a1 1 0 00-1 1v14a1 1 0 001 1z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="ml-2 flex-1 w-0 truncate"
          >
            /api/films/6/
          </span>
        </div>
        <div
          class="ml-4 flex-shrink-0"
        >
          <a
            class="font-medium text-indigo-600 hover:text-indigo-500"
            href="/api/films/6/"
          >
            Visit
          </a>
        </div>
      </li>
    </ul>
  </div>
</body>
`;
